File: src/test/resources/oracle/commits/rm-action-test-8187d840c38c0a1eaf8617c1290f2782084dd059/OrderProcessor.java
public class OrderProcessor {
    private final double taxRate = 0.08;

    public String buildReceipt(String customerName, int unitPrice, int quantity, int discount) {
        int subtotal = unitPrice * quantity;
        int discountedSubtotal = subtotal - discount;
        int tax = (int) (discountedSubtotal * taxRate);
        int total = discountedSubtotal + tax;

        String header = formatHeader(customerName);
        double dollars = centsToDollars(total);
        String status = formatPaymentStatus(total, discount);

        return header + "\n"
                + "Subtotal: " + subtotal + "\n"
                + "Discount: " + discount + "\n"
                + "Tax: " + tax + "\n"
                + "Total: $" + dollars + "\n"
                + status;
    }

    private String formatHeader(String customerName) {
        return "Receipt for " + customerName.trim().toUpperCase();
    }

    private double centsToDollars(int cents) {
        return cents / 100.0;
    }

    private String formatPaymentStatus(int total, int discount) {
        if (total > 0 && discount > 0) {
            return "Discount applied";
        }
        if (total > 0) {
            return "Payment required";
        }
        return "No payment required";
    }
}

File: src/test/resources/oracle/commits/rm-action-test-0a5085070f80f44c03645e1557c2bba8da3b335e/typescript/OrderProcessor.ts
export class OrderProcessor {
    private readonly taxRate = 0.08;

    buildReceipt(customerName: string, unitPrice: number, quantity: number, discount: number): string {
        const subtotal = unitPrice * quantity;
        const discountedSubtotal = subtotal - discount;
        const tax = Math.floor(discountedSubtotal * this.taxRate);
        const total = discountedSubtotal + tax;

        const header = this.formatHeader(customerName);
        const dollars = this.centsToDollars(total);
        const status = this.formatPaymentStatus(total, discount);

        return (
            `${header}\n` +
            `Subtotal: ${subtotal}\n` +
            `Discount: ${discount}\n` +
            `Tax: ${tax}\n` +
            `Total: $${dollars}\n` +
            `${status}`
        );
    }

    private formatHeader(customerName: string): string {
        return "Receipt for " + customerName.trim().toUpperCase();
    }

    private centsToDollars(cents: number): number {
        return cents / 100.0;
    }

    private formatPaymentStatus(total: number, discount: number): string {
        if (total > 0 && discount > 0) {
            return "Discount applied";
        }
        if (total > 0) {
            return "Payment required";
        }
        return "No payment required";
    }
}
